src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -920,68 +1087,75 @@ avtUnstructuredDomainBoundaries::ExchangeMaterial(vector<int>    domainNum,
 //
 //    Eric Brugger, Fri Mar 13 15:20:08 PDT 2020
 //    Modify to handle NULL meshes.
+// 
+//    Justin Privitera, Wed Apr 23 17:39:24 PDT 2025
+//    Style updates, changed index variable names to be more descriptive.
+//    Added const where possible.
+//    Use a map of class objects instead of pointers.
+//    Use STL containers where possible.
+//    Use std::copy instead of memcpy.
 //
 // ****************************************************************************
 
 vector<avtMaterial*>
 avtUnstructuredDomainBoundaries::ExchangeMixedMaterials(vector<int> domainNum,
-                                                     vector<avtMaterial*> mats)
+                                                        vector<avtMaterial*> mats)
 {
-    int   ***matlist = NULL;
-    int   ***mixm = NULL;
-    float ***mixvf = NULL;
-
     vector<int> domain2proc = CreateDomainToProcessorMap(domainNum);
 
-    int **nGainedCells = NULL;
-    int **nGainedMixlen = NULL;
-    CommunicateMaterialInformation(domain2proc, domainNum, mats,
-                                   nGainedCells, nGainedMixlen,
-                                   matlist, mixm, mixvf);
+    std::map<int, std::map<int, MixedMaterialDomainData>> domaindata;
+    CommunicateMaterialInformation(domain2proc, domainNum, mats, domaindata);
 
     vector<avtMaterial*> out(mats.size(), NULL);
 
-    for (size_t i = 0; i < domainNum.size(); ++i)
+    for (size_t domId = 0; domId < domainNum.size(); domId ++)
```

Comment:
Again, I think the iteration loop variable is an *index*, not a domain id. `i` was fine. `domIndex` would be fine. `domId` is potentially confusing.

Reply:
I changed them all as discussed above.